oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/resource/web/BearerTokenResolver.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.resource.web;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * A strategy for resolving Bearer Token from the {@link HttpServletRequest}.
+ *
+ * @author Vedran Pavic
+ * @since 5.1
+ * @see <a href="https://tools.ietf.org/html/rfc6750#section-2" target="_blank">RFC 6750 Section 2: Authenticated Requests</a>
+ */
+public interface BearerTokenResolver {
```

Comment:
Authorization Server module that needs this code also serves in role of Resource Server (for example, OpenID Provider with UserInfo Endpoint) and should pull in `oauth2-resource-server` module. If we follow the logic to move things under `oauth2-core` because Authorization Server could use it then all of the Resource Server code might as well go into `oauth2-core`.

Reply:
Agreed - `oauth2-authorization-server` will pull in `oauth2-resource-server`